cat/linux-kernel.cat

```diff
@@ -18,27 +18,37 @@
  * It can be replaced by include "cos.cat" for tests that do not use locks.
  *)
 
-include "cos.cat"
+include "lock.cat"
+
+(* Compute marked and plain memory accesses *)
+let Marked = (~M) | IW | Once | Release | Acquire | (RMW & R) | (RMW & W) |
```

Comment:
No (`BeginAtomic` and `EndAtomic` do, but they will never show up in a litmus test). I agree that the definition could be simplified to `RMW`, but I wanted to keep it as close as possible to the original one which uses `domain(rmw) | range(raw)`

Reply:
`BeginAtomic` and `EndAtomic` are not visible though. I'm thinking more about what concrete Linux event could be RMW without being a read/write. If there was none, the original model would probably not bother with the intersection (also, why not just `RMW & M`).
Could it be related to lock events? IIRC some lock events where not treated as usual read/write events and had to be added explicitly to `rf` and `co` in `lock.cat`.
Or maybe its just a remnant of some previous experimental wmm that had additional events RMW events.

Reply:
After playing around with many modifications, I came up to the conclusion that this thing herd does about "adding some events to a set / relation later" is due to how they compute certain things and this seems not to affect us. For example, `UL` events are not supposed to initially be part of `co`, but latter they are added. In our case we add them since the beginning and I could not find any case where this cause problems.